sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3269,14 +3269,6 @@ object functions {
    */
   def raise_error(c: Column): Column = Column.fn("raise_error", c)
 
-  /**
-   * Throws an exception with the provided error class and parameter map.
-   *
-   * @group misc_funcs
-   * @since 4.0.0
-   */
-  def raise_error(c: Column, e: Column): Column = Column.fn("raise_error", c, e)
```

Comment:
We should remove this API from `connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala`

Reply:
@cloud-fan This PR will forbid to expose the `errorParms`?

Reply:
done.

@beliefer  Yes, idea is to forbid errorParams exposure, given that we will always throw USER_RAISED_EXCEPTION with single arg (string representing the message). @cloud-fan, please confirm if I got things right.